### ISSUE TITLE
Lambda /tmp full / core-dump issue

### ISSUE THREAD

**Issue by StewartSnow** [ChSankalp/Alumni#1]
Hey @ChSankalp ,
Sorry for the random issue here - just trying to get in touch to see if you worked out the solution to this problem:
https://github.com/dotnet/aspnetcore/issues/48149

The issue got closed so couldn't follow up there... We're having the exact same behaviour ourselves randomly on production. 

Anecdotally - it's something that's only cropped up for us on code running with AWS Lambda since upgrading to .net6

Any help / advice appreciated.

**Comment by ChSankalp** [comment]
Hey @StewartSnow, 

We are still experiencing the issue randomly on our production. Given that the containers for lambda spin up on runtime, we are not able to access the container's tmp storage. 

I have written an API to read the first few bytes of the files in tmp and got a few problems there as well. 

We couldn't conclude with evidence that the files belonged to the core dump and the issue got closed. I am trying to get the first few bytes and open an issue again.

--(Assuming you are getting the error for json deserialise through Newtonsoft) 

For the time being, you can stop using disk buffering for response sizes greater than 32KB (in our case) and deserialize it in memory or increase the limit. As far as I have looked into this issue, not using disk buffering wouldn't affect us very much directly by blocking the threads because lamba is a single threaded environment. 

Please let me know if you find anything new on this!

**Comment by StewartSnow** [comment]
Hey @ChSankalp ,
Thanks for all of the details - we're in a similar state, but here's what we've found / concluded thus far:

- We set the Newtsoft Json output formatter to 30MB (ie 1000 times bigger than the default) - which addressed the issue to 'some' extent, but not all. 
- We also increased the local storage from 512 MB to 1024 (still failed every so often)
- More recently we've increased the local storage to 2048 (and as yet, have not had a repeat of the issue...)

But it is still early days - and it feels like the Newtonsoft error is just a symptom of the storage being completely full, rather than Newtonsoft's cause in the first place...

Theoretically - it should continue to fail - we've not got as far as trying to read what's in the /tmp folder specifically - if we do anything it'll be likely detecting the error then somehow forcing that particular lambda container to just die.

**Comment by ChSankalp** [comment]
> it feels like the Newtonsoft error is just a symptom of the storage being completely full, rather than Newtonsoft's cause in the first place

Exactly. Newtonsoft is just trying to access the storage to buffer, which the container is unable to provide.

I am presently working on fetching the first few headers of the file in some way.
